avocado/utils/pci.py

```diff
@@ -0,0 +1,325 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM
+# Author: Narasimhan V <sim@linux.vnet.ibm.com>
+#
+# Author: Kleber Sacilotto de Souza <klebers@linux.vnet.ibm.com>
+# Author: Daniel Kreling <kreling@linux.vnet.ibm.com>
+# for get_memory_address() and get_mask()
+
+"""
+Module for all PCI devices related functions.
+"""
+
+
+import re
+import os
+from genio import read_file
+from . import process
+
+
+def get_domains():
+    """
+    Gets all pci domains.
+    Example, it returns ['0000', '0001', ...]
+
+    :return: List of pci domains.
+    """
+    cmd = "lspci -D"
+    output = process.system_output(cmd, ignore_status=True)
+    if output:
+        domains = []
+        for line in output.splitlines():
+            domains.append(line.split(":")[0])
+        return list(set(domains))
+
+
+def get_pci_addresses():
+    """
+    Gets list of pci addresses in the system.
+    Does not return the PCI Bridges/Switches.
+
+    :return: list of pci addresses.
+    """
+    addresses = []
+    cmd = "lspci -D"
+    for line in process.system_output(cmd).splitlines():
+        if "PCI bridge" not in line:
+            addresses.append(line.split()[0])
+    if addresses:
+        return addresses
+
+
+def get_num_devices_in_domain(domain):
+    """
+    Gets number of devices in a pci domain.
+
+    :parm domain: pci domain.
+
+    :return: number of devices in a pci domain.
+    """
+    cmd = "ls -l /sys/class/*/ -1"
+    output = process.system_output(cmd, ignore_status=True)
+    if output:
+        domain = '/%s/' % domain
+        count = 0
+        for line in output.splitlines():
+            if domain in line:
+                count += 1
+        return count
+
+
+def get_disks_in_pci_address(pci_address):
+    """
+    Gets disks in a pci_address.
+
+    :parm pci_address: pci address.
+
+    :return: list of disks in a pci address.
+    """
+    disks_path = "/dev/disk/by-path/"
+    disk_list = []
+    for dev in os.listdir(disks_path):
+        if pci_address in dev:
+            link = os.readlink(os.path.join(disks_path, dev))
+            disk_list.append(os.path.abspath(os.path.join(disks_path, link)))
+    return disk_list
+
+
+def get_nics_in_pci_address(pci_address):
+    """
+    Gets network interface(nic) in a pci address.
+
+    :parm pci_address: pci address.
```

Comment:
Nice, allows partial matches as well (how about adding examples to those, for example here it could be:

```
:param pci_address: pci address (eg. 0000:00:19.0 or 19.0)
```

And on places where it has to be the full path (with/without domain) the example would state that.